pkgs/tools/virtualization/ovftool/default.nix

```diff
@@ -1,149 +1,253 @@
-                    };
 in
-stdenv.mkDerivation rec {
+stdenv.mkDerivation {
   pname = "ovftool";
-  inherit version;
+  inherit (ovftoolSystem) version;
 
-  src = ovftoolSource;
+  src = fetchurl {
+    inherit (ovftoolSystem) name url hash;
+  };
 
   buildInputs = [
-    glibc
-    libxcrypt-legacy
     c-ares
     expat
     icu60
+    libiconv
+    libxcrypt-legacy
     xercesc
     zlib
+  ] ++ lib.optionals stdenv.isLinux [
+    glibc
+  ] ++ lib.optionals stdenv.isDarwin [
+    darwin.Libsystem
+    libxml2
   ];
 
-  nativeBuildInputs = [ autoPatchelfHook makeWrapper unzip ];
-
-  preferLocalBuild = true;
-
-  sourceRoot = ".";
+  nativeBuildInputs = [ unzip makeWrapper ]
+    ++ lib.optionals stdenv.isLinux [ autoPatchelfHook ];
 
-  unpackPhase = ovftoolSystem.unpackPhase;
-
-  # Expects a directory named 'ovftool' containing the ovftool install.
-  # Based on https://aur.archlinux.org/packages/vmware-ovftool/
-  # with the addition of a libexec directory and a Nix-style binary wrapper.
   installPhase = ''
     runHook preInstall
-    if [ -d ovftool ]; then
-      # Ensure we're in the staging directory
-      cd ovftool
+
+    # Based on https://aur.archlinux.org/packages/vmware-ovftool/
+    # with the addition of a libexec directory and a Nix-style binary wrapper.
+
+    # Almost all libs in the package appear to be VMware proprietary except for
+    # libgoogleurl and libcurl. The rest of the libraries that the installer
+    # extracts are omitted here, and provided in buildInputs. Since libcurl
+    # depends on VMware's OpenSSL, both libs are still used.
+    # FIXME: Replace libgoogleurl? Possibly from Chromium?
+    # FIXME: Tell VMware to use a modern version of OpenSSL. As of ovftool
+    # v4.6.2 ovftool uses openssl-1.0.2zh which in seems to be the extended
+    # support LTS release: https://www.openssl.org/support/contracts.html
+
+    # Remove packaged libs that are linked during fixup. Darwin dylibs are
+    # under lib in the zip.
+    rm -f \
+      libcares.so.2 \
+      libexpat.so \
+      libgcc_s.so.1 \
+      libicudata.so.60 \
+      libicuuc.so.60 \
+      libstdc++.so.6 \
+      libxerces-c-3.2.so \
+      libz.so.1 \
+      lib/libcares.2.dylib \
+      lib/libexpat.dylib \
+      lib/libicudata.60.2.dylib \
+      lib/libicuuc.60.2.dylib \
+      lib/libxerces-c-3.2.dylib \
+
+    # Install all remaining libs
+    install -m 755 -d "$out/lib"
+    install -m 644 -t "$out/lib" *.so *.so.* "lib/"*.dylib
+
+    # Install libexec binaries
+    # ovftool expects to be run relative to certain directories, namely `env`.
+    # Place the binary and those dirs in libexec.
+    install -m 755 -d "$out/libexec"
+    install -m 644 -t "$out/libexec" icudt44l.dat
+    install -m 755 -t "$out/libexec" "${ovftoolSystem.binName}"
+
+    # Rename to ovftool on install for all systems to ovftool
+    if [ ! -f "$out/libexec/ovftool" ]; then
+      mv "$out/libexec/${ovftoolSystem.binName}" "$out/libexec/ovftool"
```

Comment:
Instead of needing `binName`, why not just normalize the ovftool binary path at the beginning of installPhase:

```
if [ -f ovftool.bin ] && [ ! -f ovftool ]; then
    mv ovftool.bin ovftool
fi
```